New Fs Util broke Quartz and multitenancy quickstarts

`hibernate-orm-multi-tenancy-quickstart` and `quartz-quickstart` are failing with the following error:

```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:build (default) on project hibernate-orm-multi-tenancy-quickstart: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.flyway.FlywayProcessor#build threw an exception: java.nio.file.FileSystemNotFoundException
[ERROR] 	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.getFileSystem(ZipFileSystemProvider.java:169)
[ERROR] 	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.getPath(ZipFileSystemProvider.java:155)
[ERROR] 	at java.base/java.nio.file.Path.of(Path.java:208)
[ERROR] 	at io.quarkus.flyway.FlywayProcessor.initFileSystem(FlywayProcessor.java:285)
[ERROR] 	at io.quarkus.flyway.FlywayProcessor.discoverApplicationMigrations(FlywayProcessor.java:231)
[ERROR] 	at io.quarkus.flyway.FlywayProcessor.build(FlywayProcessor.java:99)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:887)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:829)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```

Failing path is `jar:file:/data/home/gsmet/git/quarkus-quickstarts/quartz-quickstart/target/quartz-quickstart-1.0.0-SNAPSHOT.jar!/db/migration/` for Quartz.

I have no idea if it's correct and should contain the `!/db/migration/` part. Also, I don't know if the `env.put("create", "true");` is of interest.

We need a fix for 2.6.0.Final so before Tuesday evening.

/cc @Postremus @aloubyansky

--- a/extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java
+++ b/extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java
@@ -4,16 +4,11 @@ import static io.quarkus.deployment.annotations.ExecutionTime.STATIC_INIT;
 
 import java.io.IOException;
 import java.lang.reflect.Modifier;
-import java.net.URI;
-import java.net.URISyntaxException;
-import java.net.URL;
-import java.nio.file.FileSystem;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.AbstractCollection;
 import java.util.Collection;
-import java.util.Enumeration;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.LinkedHashSet;
@@ -60,7 +55,7 @@ import io.quarkus.deployment.recording.RecorderContext;
 import io.quarkus.flyway.runtime.FlywayBuildTimeConfig;
 import io.quarkus.flyway.runtime.FlywayContainerProducer;
 import io.quarkus.flyway.runtime.FlywayRecorder;
-import io.quarkus.fs.util.ZipUtils;
+import io.quarkus.runtime.util.ClassPathUtils;
 
 class FlywayProcessor {
 
@@ -210,7 +205,7 @@ class FlywayProcessor {
     }
 
     private Collection<String> discoverApplicationMigrations(Collection<String> locations)
-            throws IOException, URISyntaxException {
+            throws IOException {
         LinkedHashSet<String> applicationMigrationResources = new LinkedHashSet<>();
         // Locations can be a comma separated list
         for (String location : locations) {
@@ -220,28 +215,19 @@ class FlywayProcessor {
                 continue;
             }
 
-            Enumeration<URL> migrations = Thread.currentThread().getContextClassLoader().getResources(location);
-            while (migrations.hasMoreElements()) {
-                URL path = migrations.nextElement();
-                LOGGER.infov("Adding application migrations in path ''{0}'' using protocol ''{1}''", path.getPath(),
-                        path.getProtocol());
-                final Set<String> applicationMigrations;
-                if (JAR_APPLICATION_MIGRATIONS_PROTOCOL.equals(path.getProtocol())) {
-                    try (final FileSystem fileSystem = initFileSystem(path.toURI())) {
-                        applicationMigrations = getApplicationMigrationsFromPath(location, path);
-                    }
-                } else if (FILE_APPLICATION_MIGRATIONS_PROTOCOL.equals(path.getProtocol())) {
-                    applicationMigrations = getApplicationMigrationsFromPath(location, path);
-                } else {
-                    LOGGER.warnv(
-                            "Unsupported URL protocol ''{0}'' for path ''{1}''. Migration files will not be discovered.",
-                            path.getProtocol(), path.getPath());
-                    applicationMigrations = null;
+            String finalLocation = location;
+            ClassPathUtils.consumeAsPaths(Thread.currentThread().getContextClassLoader(), location, path -> {
+                Set<String> applicationMigrations = null;
+                try {
+                    applicationMigrations = FlywayProcessor.this.getApplicationMigrationsFromPath(finalLocation, path);
+                } catch (IOException e) {
+                    LOGGER.warnv(e,
+                            "Can't process files in path %s", path);
                 }
                 if (applicationMigrations != null) {
                     applicationMigrationResources.addAll(applicationMigrations);
                 }
-            }
+            });
         }
         return applicationMigrationResources;
     }
@@ -266,9 +252,8 @@ class FlywayProcessor {
         return location;
     }
 
-    private Set<String> getApplicationMigrationsFromPath(final String location, final URL path)
-            throws IOException, URISyntaxException {
-        Path rootPath = Paths.get(path.toURI());
+    private Set<String> getApplicationMigrationsFromPath(final String location, final Path rootPath)
+            throws IOException {
 
         try (final Stream<Path> pathStream = Files.walk(rootPath)) {
             return pathStream.filter(Files::isRegularFile)
@@ -278,10 +263,6 @@ class FlywayProcessor {
                     .peek(it -> LOGGER.debugf("Discovered path: %s", it))
                     .collect(Collectors.toSet());
         }
-    }
-
-    private FileSystem initFileSystem(final URI uri) throws IOException {
-        return ZipUtils.newZip(Path.of(uri));
     }
 
     /**

